google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/facet/impl/AppEngineJavaeeSupportContributor.java

```diff
@@ -68,7 +68,7 @@ private void setWebXmlServletVersion(Module module) {
     StartupManager.getInstance(module.getProject()).runWhenProjectIsInitialized(() -> {
       for (WebFacet webFacet : WebFacet.getInstances(module)) {
         final WebApp webApp = webFacet.getRoot();
-        if (webApp == null) {
+        if (webApp == null || !webApp.getXmlTag().isWritable()) {
```

Comment:
It should be enough, I don't know when/why this would return true but the write command would still fail due to a read-only modification exception.

Good point about the test, let me add that now.